mitmproxy/master.py

```diff
@@ -32,16 +33,25 @@ class Master:
     """
         The master handles mitmproxy's main event loop.
     """
-    def __init__(self, opts, server):
-        self.options = opts or options.Options()
+    def __init__(self, opts):
+        self.options = opts or options.Options()  # type: options.Options
         self.commands = command.CommandManager(self)
         self.addons = addonmanager.AddonManager(self)
         self.event_queue = queue.Queue()
         self.should_exit = threading.Event()
-        self.server = server
+        self._server = None
         self.first_tick = True
-        channel = controller.Channel(self.event_queue, self.should_exit)
-        server.set_channel(channel)
+
+    @property
+    def server(self):
+        return self._server
+
+    @server.setter
+    def server(self, server):
+        server.set_channel(
+            controller.Channel(self.event_queue, self.should_exit)
+        )
+        self._server = server
```

Comment:
This is a bit ugly, but the plan is to ultimately get rid of `master.server` anyways, so...